examples/flutter_gallery/lib/demo/material/chip_demo.dart

```diff
@@ -12,43 +76,241 @@ class ChipDemo extends StatefulWidget {
+        onSelected: tools.contains(name)
+            ? (bool value) {
+                setState(() {
+                  if (!value) {
+                    selectedTools.remove(name);
+                  } else {
+                    if (!selectedTools.contains(name)) {
+                      selectedTools.add(name);
+                    }
+                  }
+                });
+              }
+            : null,
+      );
+    }).toList();
+
+    Set<String> allowedActions = new Set<String>();
+    if (selectedMaterial != null && selectedMaterial.isNotEmpty) {
+      for (String tool in selectedTools) {
+        allowedActions = allowedActions.union(toolActions[tool]);
+      }
+      allowedActions = allowedActions.intersection(materialActions[selectedMaterial]);
     }
 
+    final List<Widget> actionChips = allowedActions.map<Widget>((String name) {
+      return new ActionChip(
+        label: new Text(_capitalize(name)),
+        onPressed: () {
+          setState(() {
+            selectedAction = name;
+          });
+        },
+      );
+    }).toList();
+
+    final ThemeData theme = Theme.of(context);
+    final List<Widget> tiles = <Widget>[
+      const SizedBox(height: 8.0, width: 0.0),
+      _wrapChips('Available Materials (Chip)', chips, theme),
+      _wrapChips('Available Tools (InputChip)', inputChips, theme),
+      _wrapChips('Choose a Material (ChoiceChip)', choiceChips, theme),
+      _wrapChips('Choose Tools (FilterChip)', filterChips, theme),
+      _wrapChips('Perform Allowed Action (ActionChip)', actionChips, theme),
+      const Divider(),
+      new Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: new Center(
+          child: new Text(
+            _createResult(),
+            style: Theme.of(context).textTheme.title,
+          ),
+        ),
+      ),
+    ];
+
     return new Scaffold(
-      appBar: new AppBar(title: const Text('Chips')),
-      body: new ListView(
-        children: chips.map((Widget chip) {
-          return new Container(
-            height: 100.0,
-            child: new Center(child: chip)
-          );
-        }).toList()
-      )
+      appBar: new AppBar(
+        title: const Text('Chips'),
+        actions: <Widget>[
+          new IconButton(
+            onPressed: () {
+              setState(() {
+                showShapeBorder = !showShapeBorder;
+              });
+            },
+            icon: const Icon(Icons.vignette),
+          )
+        ],
+      ),
+      body: new ChipTheme(
+        data: showShapeBorder
+            ? Theme.of(context).chipTheme.copyWith(
+                    shape: new BeveledRectangleBorder(
+                  side: const BorderSide(width: 0.66, style: BorderStyle.solid, color: Colors.grey),
+                  borderRadius: new BorderRadius.circular(10.0),
+                ))
+            : Theme.of(context).chipTheme.copyWith(shape: const StadiumBorder()),
```

Comment:
Shouldn't this just be the default chip theme?

Reply:
Yeah, I guess so.